From 280c921903f9cea4eec1a725ebd3462c17080e48 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 23 Apr 2024 14:22:35 +0200
Subject: [PATCH 0839/1695] plat-stm32mp1: conf: move definitions

Move configuration directives definition in plat-stm32mp1/conf.mk
to ease introduction of CFG_STM32MP_PROFILE that is used to
set predefined values to some CFG_xxx configuration directives.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a45cdf1ab5c53838996c8025e2f0fcc01819de0f plat-stm32mp1: conf: move definitions]
Change-Id: I464665a62f0e3f5855601242fcc94c324a8f35c8
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/376064
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 46 +++++++++++++++--------------
 1 file changed, 24 insertions(+), 22 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index d2ffb266d..06b6bdd7c 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -104,41 +104,24 @@ CFG_EMBED_DTB_SOURCE_FILE ?= $(flavor_dts_file-$(PLATFORM_FLAVOR))
 endif
 CFG_EMBED_DTB_SOURCE_FILE ?= stm32mp157c-dk2.dts
 
-ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_cryp)),)
-$(call force,CFG_STM32_CRYP,n)
-$(call force,CFG_STM32_SAES,n)
-endif
-
-ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_rng)),)
-$(call force,CFG_HWRNG_PTA,n)
-$(call force,CFG_WITH_SOFTWARE_PRNG,y)
-endif
-
-ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP15-HUK-DT)),)
-CFG_STM32MP15_HUK ?= y
-CFG_STM32_HUK_FROM_DT ?= y
-endif
-
+# CFG_STM32MP1x switches are exclusive.
+# - CFG_STM32MP15 is enabled for STM32MP15x-* targets (default)
+# - CFG_STM32MP13 is enabled for STM32MP13x-* targets
+# We try to guess the variant from the embedded DT source file name
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP13)),)
 $(call force,CFG_STM32MP13,y)
 endif
-
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP15)),)
 $(call force,CFG_STM32MP15,y)
 endif
 
-# Default do not access external DT passed to non-secure boot stage
-CFG_EXTERNAL_DT ?= n
-
-# CFG_STM32MP1x switches are exclusive.
-# - CFG_STM32MP15 is enabled for STM32MP15x-* targets (default)
-# - CFG_STM32MP13 is enabled for STM32MP13x-* targets
 ifeq ($(CFG_STM32MP13),y)
 $(call force,CFG_STM32MP15,n)
 else
 $(call force,CFG_STM32MP15,y)
 $(call force,CFG_STM32MP13,n)
 endif
+
 ifeq ($(call cfg-one-enabled,CFG_STM32MP15 CFG_STM32MP13),n)
 $(error One of CFG_STM32MP15 CFG_STM32MP13 must be enabled)
 endif
@@ -146,6 +129,25 @@ ifeq ($(call cfg-all-enabled,CFG_STM32MP15 CFG_STM32MP13),y)
 $(error Only one of CFG_STM32MP15 CFG_STM32MP13 must be enabled)
 endif
 
+# Some platform do not support some resources
+ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP15-HUK-DT)),)
+CFG_STM32MP15_HUK ?= y
+CFG_STM32_HUK_FROM_DT ?= y
+endif
+
+ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_cryp)),)
+$(call force,CFG_STM32_CRYP,n)
+$(call force,CFG_STM32_SAES,n)
+endif
+
+ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_rng)),)
+$(call force,CFG_HWRNG_PTA,n)
+CFG_WITH_SOFTWARE_PRNG ?= y
+endif
+
+# Default do not access external DT passed to non-secure boot stage
+CFG_EXTERNAL_DT ?= n
+
 include core/arch/arm/cpu/cortex-a7.mk
 
 $(call force,CFG_ARM_GIC_PM,y)
-- 
2.47.2

